python/pylibcudf/pylibcudf/types.pyx

```diff
@@ -138,6 +182,47 @@ cdef class DataType:
         ret.c_obj = dt
         return ret
 
+    @staticmethod
+    def from_arrow(pa_typ) -> DataType:
+        """
+        Construct a DataType from a Python type.
+
+        Parameters
+        ----------
+        pa_typ : pyarrow type
+            A Pyarrow type (eg. pa)
+
+        Returns
+        -------
+        DataType
+            The corresponding pylibcudf DataType.
+
+        Raises
+        ------
+        ImportError
+            If pyarrow is not installed.
+        TypeError
+            If the Python type is not supported.
+        """
+        if pa_err is not None:
+            raise pa_err
+        return DataType._from_arrow(pa_typ)
+
+    if pa is not None:
+        @staticmethod
+        def _from_arrow(obj: pa.DataType) -> DataType:
+            if isinstance(obj, pa.Decimal128Type):
```

Comment:
Could you add support for `pa.Decimal32/64Type` if pyarrow 19 is available (or use the `getattr(pa, "Decimal32/64Type", None)` trick to see if it's available)

Reply:
Added some tests too https://github.com/rapidsai/cudf/pull/18938/commits/a40e06ea5471f63f89cced65dd46662429024884